### ISSUE TITLE
Missing a .meta file for the mujoco.dll in the unity plugin

### ISSUE THREAD

**Issue by ganyuanzhen** [google-deepmind/mujoco#1146]
I'm using MuJoCo and Unity on Windows 11. However, when I imported the package via git in the Unity editor, it promoted that the mujoco.dll does not have a .meta file and is ignored. 

After I create a .meta file for mujoco.dll manually, the plugin loads successfully. 

So, I think it is better to add .meta files for the runtime libraries.

**Comment by ganyuanzhen** [comment]
Or, it is better to create a meta file for the library while copying the library during the installation process.

**Comment by Balint-H** [comment]
How you describe it is one way to do it, although the copied files may disappear from the package folder if the project needs to reimport the git package. The way I usually handle this is to copy the mujoco DLL straight to my Project (somewhere in the Assets folder). 

To better support git based installations, maybe the install instructions could be updated with this approach?

Alternatively, maybe including the dll and meta file in the package itself could be appropriate. Or is including a dll like that not good practice?

**Comment by ganyuanzhen** [comment]
> How you describe it is one way to do it, although the copied files may disappear from the package folder if the project needs to reimport the git package. The way I usually handle this is to copy the mujoco DLL straight to my Project (somewhere in the Assets folder).
> 
> To better support git based installations, maybe the install instructions could be updated with this approach?
> 
> Alternatively, maybe including the dll and meta file in the package itself could be appropriate. Or is including a dll like that not good practice?

Including a dll file in the repo doesn't seem to be a  very well practise, although it is convenient for users to import. Doing so will lead to the increment of the repo's size at every update, as the dll is a binary file. 

As indicated in the ![document](https://mujoco.readthedocs.io/en/stable/unity.html), "On Unity version 2020.2 and later, the Package Manager will look for the native library file and copy it to the package directory when the package is imported." But the Package Manager will only copy the library file itself, **not** creating the .meta file at the same time. I think this will be the major problem. 

P.S. I use Unity 2020.3.48f1c1. I'm not sure whether on higher version of Unity this issue will occur or not.

Therefor, the install instructions should be updated to indicate this.

**Comment by ganyuanzhen** [comment]
I already open a pr to fix the problem.

**Comment by GF-Huang** [comment]
Still for me today. I install from git URL: `https://github.com/google-deepmind/mujoco.git?path=/unity#3.6.0`

<img width="1152" height="544" alt="Image" src="https://github.com/user-attachments/assets/9fe30db3-1ee7-4553-a95d-aeac910a7b45" />

<img width="1541" height="420" alt="Image" src="https://github.com/user-attachments/assets/7691a538-106e-4a7f-87b4-1e81651c4bf1" />

**Comment by Balint-H** [comment]
Elysia's solution https://github.com/google-deepmind/mujoco/pull/1154 is a reasonable fix for this but was never merged. I think it might be a better pattern to not do any copying of the binaries but to leave it up to the user to supply the library to their project (or use a solution like @joanllobera's https://github.com/joanllobera/mujoco-bin)

This error does not block the plugin from working and can be safely cleared. However of course it would be neater.

@GF-Huang what do you think, is automatic searching and copying of the binaries a good feature? Or would it be fine if the user needed to install the binaries to the project?

**Comment by GF-Huang** [comment]
Most users expect it to work out of the box, while more advanced users should be able to easily dive into the source code. A reasonable design accommodates both.

**Comment by joanllobera** [comment]
I don't believe we should assume that naive users will be on windows. They can as well be on mac, or even on linux.

If what we want, particularly for naive users, is to work out of the box, it would be easier to have a small package resolving for each platform. It is with this thought that I did this package: https://github.com/joanllobera/mujoco-bin 

That contains the binaries and the correct configuration in the .meta files so they are properly imported in unity.
If users know how to import the mujoco package from the git url, they will also know how to import the mujoco-bin package, with the same procedure. 

Of course, though, the mujoco-bin package needs to be updated with each release.  But that is not a lot of work, it is just a matter of replacing the binary files, committing, and adding a tag that matches the mujoco release.

**Comment by GF-Huang** [comment]
@joanllobera If add [joanllobera/mujoco-bin](https://github.com/joanllobera/mujoco-bin?rgh-link-date=2026-03-25T15%3A40%3A41.000Z) to MuJoCo official docs will be better for new user.
